src/services/redeem_and_unstake/AcceptRedeemRequestService.ts

```diff
@@ -0,0 +1,337 @@
+
+    const rawTx: TransactionObject<string> = redeemPool.methods.acceptRedeemRequest(
+      (redeemRequest.amount as BigNumber).toString(10),
+      (redeemRequest.beneficiary as string),
+      (redeemRequest.gasPrice as BigNumber).toString(10),
+      (redeemRequest.gasLimit as BigNumber).toString(10),
+      (redeemRequest.nonce as BigNumber).toString(10),
+      (redeemRequest.sender as string),
+      (redeemRequest.gateway as string),
+      hashLock,
+    );
+    const bounty = await this.getBountyAmount(redeemRequest);
+    return Utils.sendTransaction(rawTx, {
+      from: this.auxiliaryWorkerAddress,
+      gasPrice: AUXILIARY_GAS_PRICE,
+      value: bounty,
+    }, this.web3);
+  }
+
+  /**
+   * Creates entry in message repository.
+   * @param redeemRequest Redeem request object.
+   * @param secret Secret of the redeem request.
+   * @param hashLock Hash lock for the redeem request.
+   * @return Returns message hash.
+   */
+  private async createMessageInRepository(
+    redeemRequest: MessageTransferRequest,
+    secret: string,
+    hashLock: string,
+  ): Promise<string> {
+    const messageHash = this.calculateMessageHash(redeemRequest, hashLock);
+    Logger.debug(`Creating message for message hash ${messageHash}`);
+
+    const message = new Message(
+      messageHash,
+      MessageType.Redeem,
+      redeemRequest.gateway as string,
+      MessageStatus.Undeclared,
+      MessageStatus.Undeclared,
+      redeemRequest.gasPrice as BigNumber,
+      redeemRequest.gasLimit as BigNumber,
+      redeemRequest.nonce as BigNumber,
+      redeemRequest.senderProxy as string,
+      MessageDirection.AuxiliaryToOrigin,
+      new BigNumber(0),
+      secret,
+      hashLock,
+    );
+
+    await this.messageRepository.save(message);
+    Logger.debug(`Message object saved for message hash ${message.messageHash}`);
+    return messageHash;
+  }
+
+  /**
+   * Updates the message hash in messageTransferRequest repository after
+   * accepting redeem request. Accepting redeem requests adds a new entry
+   * into messages' repository with a message hash. That exact message
+   * hash is updated here in messageTransferRequest repository.
+   */
+  private async updateMessageHash(
+    redeemRequestHash: string,
+    messageHash: string,
+    blockNumber: BigNumber,
+  ): Promise<void> {
+    const redeemRequest = new MessageTransferRequest(
+      redeemRequestHash,
+      RequestType.Redeem,
+      blockNumber,
+    );
+    redeemRequest.messageHash = messageHash;
+    Logger.debug('Updating message hash in message transfer request repository');
+    await this.messageTransferRequestRepository.save(redeemRequest);
+  }
+
+  /**
+   * It pre-calculates message hash.
+   * @param redeemRequest Redeem request object.
+   * @param hashLock Hash lock of acceptRedeem transaction.
+   * @return Returns message hash.
+   */
+  private calculateMessageHash(redeemRequest: MessageTransferRequest, hashLock: string): string {
```

Comment:
this method seems to be an exact replica of one in accept stake request service. both should come from a util class may be. similar for `generateSecret`

Reply:
refactored to Utils.ts